.travis.yml

```diff
@@ -5,74 +5,112 @@ env:
   global:
     - LD_PRELOAD=/lib/x86_64-linux-gnu/libSegFault.so
     - SEGFAULT_SIGNALS=all
-  matrix:
-    - TOXENV=check
-    - TOXENV=docs
-matrix:
+stages:
+  - lint
+  - examples
+  - tests
+jobs:
   include:
-    - python: '2.7'
+    - stage: lint
+      env:
+        - TOXENV=check
+    - stage: lint
+      env:
+        - TOXENV=docs
+    - stage: tests
+      python: '2.7'
       env:
         - TOXENV=py27-t310-c45
-    - python: '2.7'
+    - stage: tests
+      python: '2.7'
       env:
         - TOXENV=py27-t40-c45
-    - python: '2.7'
+    - stage: tests
+      python: '2.7'
```

Comment:
You do not have to repeat the stage names - I'd split them into sections separated by a newline.
Requires extra logic in the template generator though, so likely not worth it.

The generated files (.travis.yml etc) should have a comment about them being autogenerated though btw, since I've previously missed this myself.

Reply:
https://github.com/pytest-dev/pytest-cov/pull/278